packages/atomic/src/components/ipx/atomic-ipx-button/atomic-ipx-button.tsx

```diff
@@ -45,7 +46,24 @@ export class AtomicIPXButton implements InitializableComponent {
    */
   @Prop({mutable: true, reflect: true}) public isModalOpen = false;
 
+  private recommendationsLoaded = false;
+
+  private async loadRecommendations() {
+    const recsEngine = this.bindings.interfaceElement.querySelector(
+      'atomic-recs-interface'
+    )?.engine;
+    if (recsEngine) {
+      this.recommendationsLoaded = true;
+      recsEngine.dispatch(
+        loadRecommendationActions(recsEngine).getRecommendations()
+      );
+    }
```

Comment:
@louis-bompart do you think this is legit?

Reply:
Confused, why would the engine be not there?

Reply:
It should always be there. We always put the recs engine in the IPX in the SIS and we are the only users of these components

Reply:
so why the if?

Reply:
Just because the addition of the recs engine kinda requires a manual step since it has to be in the html. 

But my question was more: is it okay to put a dependency to the `recs-engine` inside the `ipx-button` component. I think it is, since the IPX components are more of an internal lib of which the SIS is the only user, so we can make it work the way we like. From your comments I understand you don't have any issues with it, so I'll open the PR.

Reply:
It will increase the size of the file that you need to import to instantiate the button. But if, as owner of the IPX, you're fine with this, eh, your call :D